src/main/java/com/example/demo/services/BookingService.java

```diff
@@ -0,0 +1,272 @@
+        );
+
+        // empty room without seats for tests
+        RoomEntity room3 = roomRepository.save(new RoomEntity(17L, 2L, new ArrayList<SeatEntity>(List.of())));
+
+
+        ReservationEntity reservation1 = reservationRepository.save(
+                new ReservationEntity(18L, "Jan", "Tracz", seat1room1.getId())
+        );
+
+        ReservationEntity reservation2 = reservationRepository.save(
+                new ReservationEntity(19L, "Anna", "Kowaluk-Łysoń", seat4room1.getId())
+        );
+
+        ReservationEntity reservation3 = reservationRepository.save(
+                new ReservationEntity(20L, "Michau", "Białek", seat5room2.getId())
+        );
+
+
+        Date date1 = new Date(1651064400000L);
+        Date date2 = new Date(1651072500000L);
+        Date date3 = new Date(1651245300000L);
+        Date date4 = new Date(1651256100000L);
+
+        MovieEntity movie1 = movieRepository.save(
+                new MovieEntity(21L, "Wyścig z czasem", date1, date2, room1,
+                        new ArrayList<ReservationEntity>(List.of(reservation1, reservation2)))
+        );
+        MovieEntity movie2 = movieRepository.save(
+                new MovieEntity(22L, "Jaś Fasola", date3, date4, room1,
+                        new ArrayList<ReservationEntity>(List.of()))
+        );
+        MovieEntity movie3 = movieRepository.save(
+                new MovieEntity(23L, "Łasuch", date1, date2, room2,
+                        new ArrayList<ReservationEntity>(List.of(reservation3)))
+        );
+        MovieEntity movie4 = movieRepository.save(
+                new MovieEntity(24L, "Łasuch", date1, date2, room3,
+                        new ArrayList<ReservationEntity>(List.of()))
+        );
+
+
+    }
+
+    public List<MovieModel> moviesInTimeInterval(Date start, Date end) {
+
+        List<MovieModel> list = movieMapper.toModel(
+                        movieRepository.findByStartBeforeAndEndAfterOrderByStart(end, start))
+                .stream()
+                .map(Optional<MovieModel>::get)
+                .collect(Collectors.toList());
+
+        return list;
+    }
+
+
+    public Optional<MovieDetails> movieDetails(Long id) {
+        MovieModel movie = movieMapper.toModel(movieRepository.findById(id).orElseThrow()).orElseThrow();
+
+        List<SeatModel> allSeats = movie.getRoom().getSeats();
+
+        List<SeatModel> availableSeats = allSeats.stream()
+                .filter(item -> !movie.getReservations().contains(item.getId()))
+                .collect(Collectors.toList());
+
+        return Optional.of(new MovieDetails(movie, availableSeats));
+    }
+
+
+    private List<SeatEntity> deepCopySeatList(List<SeatEntity> s) {
+        ArrayList<SeatEntity> result = new ArrayList<>();
+        for (SeatEntity seat : s) {
+            result.add(new SeatEntity(seat));
+        }
+        return result;
+    }
+
+    public Optional<ReservationResponseModel> makeReservation(ReservationRequestModel request) {
```

Comment:
czy to nie za długa metoda?

Reply:
jest długa, i niektóre części można by wydzielić jako funkcje pomocnicze, np. fory sprawdzające rozłożenie miejsc